third_party/move/move-vm/types/src/natives/function.rs

```diff
@@ -43,6 +44,24 @@ pub enum NativeResult {
     OutOfGas {
         partial_cost: InternalGas,
     },
+    /// Instruct the VM to perform a control flow transfer.
+    ///
+    /// Note the calling convention here requires the following:
+    /// The native function that performs the dispatch should have the same type signature as the dispatch target function except
+    /// the native function will have an extra argument in the end to determine which function to jump to.
+    ///
+    /// Failing to follow this convention will result in errors in paranoid mode.
```

Comment:
Could it somehow happen that the paranoid mode doesn't detect the error right away. For instance if all the types on the stack are of the same type.